services/auth/token/src/lambdas/generateToken.ts

```diff
@@ -1,18 +1,40 @@
 import to from 'await-to-js';
-import config from '../../../../config';
-import secrets from '../../../../libs/secrets';
-import * as response from '../../../../libs/response';
-import { signToken, verifyToken } from '../../../../libs/token';
+import config from '../libs/config';
+import secrets from '../libs/secrets';
+import * as response from '../libs/response';
+import { signToken, verifyToken } from '../libs/token';
 import { throwError } from '@helsingborg-stad/npm-api-error-handling';
 import tokenValidationSchema from '../helpers/schema';
-import log from '../../../../libs/logs';
+import log from '../libs/logs';
+import Joi from 'joi';
 
 const CONFIG_AUTH_SECRETS = config.auth.secrets;
 const ACCESS_TOKEN_EXPIRES_IN_MINUTES = 20;
 const REFRESH_TOKEN_EXPIRES_IN_MINUTES = 30;
 
-export const main = async (event, context) => {
-  const [parseJsonError, parsedJson] = await to(parseJson(event.body));
+interface SecretsConfig {
+  name: string;
+  keyName: string;
+}
+
+interface AuthLambdaRequest {
+  grant_type: string;
+  refresh_token?: string;
+  code?: string;
+}
+
+interface JWTToken {
+  personalNumber: string;
+}
+
+interface SDKError {
+  code: string;
+  message: string;
+}
+export const main = async (event: { body: string }, context: { awsRequestId: string }) => {
```

Comment:
A decision was made to declare functions as normal functions. We use arrow functions for callbacks, for example. map, filter.

We considered it more clear and legible in the code what was a function in this way.

Just like here, there are more places to fix this so it becomes consistent through the code base. It can therefore be good to correct this when we have the code in front of us. :)

Reply:
fixed